### PR TITLE
fix definition of ovh_iploadbalancing_ssl

### PR DESCRIPTION
**Pull request by beluc**
# Description

ovh_iploadbalancing_ssl: Changing service_name, certificate, key or chain must trigger a full replace

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

- [x] `make testacc TESTARGS="-run TestAccIpLoadbalancingSsl_basic"`
- [x] `make testacc TESTARGS="-run TestAccIpLoadbalancingSsl_importBasic"`

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings or issues
- [x] I have added acceptance tests that prove my fix is effective or that my feature works
- [x] New and existing acceptance tests pass locally with my changes
- [x] I ran successfully `go mod vendor` if I added or modify `go.mod` file

### CODE REVIEW

**Comment by tduboys** [comment]
There is still an issue on imported certificates, as the API only returns fingerprint, not the cert/chain/key values
So if I import my resource, the local state does not contain the cert/chain/key, and on next run, terraform compares my values to null ones on the state, not with the fingerprint, and triggers a replacement.

some ideas to fix : 
- do not compare certs given on the HCL code, but a generated fingerprint to the local known fingerprint to trigger replacement (probably the better but need ssl lib)
- set cert/chain/key as optional, but trigger an error if not provided for a create operation only
- patch iplb api to return all the datas, not only the fingerprint

**Comment by beluc** [comment]
maybe we can deal with fingerprint using https://developer.hashicorp.com/terraform/plugin/framework/resources/plan-modification#requiresreplaceif

File: website/docs/r/iploadbalancing_ssl.html.markdown [hunk 1]
```diff
@@ -20,6 +20,11 @@ resource "ovh_iploadbalancing_ssl" "sslname" {
   certificate  = "..."
   key          = "..."
   chain        = "..."
+
+  # use this if ssl is configured as frontend default_ssl
```
[the remark this answers is not in the document]
Comment (tduboys) [reply website/docs/r/iploadbalancing_ssl.html.markdown:24]: > is this really needed now that you added several `RequireReplace` on the fields ?

As we cant remove a ssl object if it's used as default ssl on a frontend, the create_before_destroy option is needed

File: ovh/resource_iploadbalancing_ssl_gen.go [hunk 2]
```diff
@@ -198,3 +229,52 @@ func (v IploadbalancingSslModel) ToUpdate() *IploadbalancingSslWritableModel {
...
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if buf.String() == data.Fingerprint.ValueString() {
+		// This ssl was created from a CERT/KEY/CHAIN but it's gone from the state.
+		// This happen when we import this ressource,
```
Comment (deathiop) [inline ovh/resource_iploadbalancing_ssl_gen.go:274]:
```suggestion
		// This happens when we import this ressource,
```

**Review by amstuta — APPROVED** [review]

**Review by amstuta — APPROVED** [review]

**Pull request merged**